source/exe/terminate_handler.cc

```diff
@@ -0,0 +1,16 @@
+#include "exe/terminate_handler.h"
+
+#include <cstdlib>
+
+#include "server/backtrace.h"
+
+namespace Envoy {
+
+std::terminate_handler TerminateHandler::logOnTerminate() const {
+  return std::set_terminate([]() {
+    BACKTRACE_LOG();
```

Comment:
Once you look at the addresses it's obvious that the backtrace and abort are due to terminate(), but that takes a bit of extra effort. How about an extra log statement here indicating that an uncaught exception occurred? That'll make it very obvious. Unfortunately the C++ standard doesn't allow the terminate handler to figure out useful things like the type of the exception or other exception details...